### ISSUE TITLE
Problems connecting to storage

### ISSUE THREAD

**Issue by gadgetlabs** [ViDA-NYU/ache#1]
I'm trying to kick ache off. I've built a model and then am using the following to run - 
```
./ache startCrawl /tmp/output /home/user/Desktop/topic/topic_config /home/user/Desktop/topic/topic.seeds /home/user/Desktop/topic/topic_model /home/user/Downloads/langdetect-03-03-2014/profiles
```
I get the same issue on ache when I check it out from continuum.io's repo. The issue is not dissimilar to this one https://github.com/memex-explorer/memex-explorer/issues/357

The following issue happens, and keeps happening. Is there anything I need to configure with regards to configuration? 

I've used the config from your repo, with my seeds and my model.

Any ideas?

```
Erro de comunicacao: Connection refused
Dormindo 5 mls
[11/MAR/2015:13:56:19] [SocketAdapterFactory] [produce] [localhost:1988]
Erro de comunicacao: Connection refused
crawler_group_2_0:Connection refused
focusedCrawler.crawler.CrawlerException: crawler_group_2_0:Connection refused
	at focusedCrawler.crawler.CrawlerImpl.selectUrl(CrawlerImpl.java:250)
	at focusedCrawler.crawler.Crawler.run(Crawler.java:258)
java.net.ConnectException: Connection refused
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:339)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:200)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:182)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:579)
	at java.net.Socket.connect(Socket.java:528)
	at java.net.Socket.<init>(Socket.java:425)
	at java.net.Socket.<init>(Socket.java:208)
	at focusedCrawler.util.storage.socket.StorageRemoteAdapter.getSocket(StorageRemoteAdapter.java:78)
	at focusedCrawler.util.storage.socket.StorageRemoteAdapter.defaultMethod(StorageRemoteAdapter.java:180)
	at focusedCrawler.util.storage.socket.StorageRemoteAdapter.select(StorageRemoteAdapter.java:300)
	at focusedCrawler.util.storage.distribution.StorageRemoteAdapterReconnect.select(StorageRemoteAdapterReconnect.java:265)
	at focusedCrawler.crawler.CrawlerImpl.selectUrl(CrawlerImpl.java:213)
	at focusedCrawler.crawler.Crawler.run(Crawler.java:258)
R>crawler_group_2_0>Sleeping 5000 mls.
RM>crawler_group>crawler_group_0_0>Time(3708):Sleeping as a consequence of this problem: 'crawler_group_0_0:Connection refused':selectUrl() linkStorage.
RM>crawler_group>crawler_group_1_0>Time(3694):Sleeping as a consequence of this problem: 'crawler_group_1_0:Connection refused':selectUrl() linkStorage.
RM>crawler_group>crawler_group_2_0>Time(3673):Sleeping as a consequence of this problem: 'crawler_group_2_0:Connection refused':selectUrl() linkStorage.
RM>crawler_group>crawler_group_3_0>Time(3707):Sleeping as a consequence of this problem: 'crawler_group_3_0:Connection refused':selectUrl() linkStorage.
RM>crawler_group>crawler_group_4_0>Time(3704):Sleeping as a consequence of this problem: 'crawler_group_4_0:Connection refused':selectUrl() linkStorage.
R>crawler_group_0_0>Total time is 5007 mls [5000,0,0,0,0,0,0,5000,0]
```

**Comment by brittainhard** [comment]
Glad someone else is coming across this problem. I was able to fix this problem on my local machine by using proper model files provided by our data scientist Christine. Unfortunately, this did not work in the case of our deployed servers, which are still suffering from this problem, despite using the same model files that I used on my local machine.

I'll talk with Christine today and see what the status is of our ache debugging, and I may be able to tackle this problem freshly by creating new model files on our deployment server. 

I'll let you know how it goes, since we are in the same boat :).

**Comment by aecio** [comment]
I'm also trying to reproduce this error and understand what is going on. I'll let you know if I find something.

**Comment by brittainhard** [comment]
So we managed to fix this on our deployment box by redeploying it on a new machine. We never managed to figure out why it was failing, and were never able to reproduce the bug. Wish I could help you guys out.

**Comment by aecio** [comment]
What version are you running? Is it the version from branch master of this repository?

Here's what I found so far about this problem. The ACHE crawler has three main services: `TargetStorage`, `LinkStorage` and `CrawlerManager`. This exception happens when the `TargetStorage` tries to communicate with the `LinkStorage`, but the `LinkStorage` is not running.

When running the crawler using the command `ache startCrawl`, all services are started at the same time in different threads of the same process. Sometimes the `TargetStorage` tries to connect to `LinkStorage`, but the `LinkStorage` has not finished initializing yet, so this exception happens. As the `TargetStorage` tries to reconnect automatically, everything works fine when the `LinkStorage` is ready. In this case, this is not a real problem, but the logs are misleading.

When each service is started in a different process, if the `LinkStorage` dies the same problem will happen. You need to restart the `LinkStorage` process in this case.

I've been working on improving the logs and detect any problems that may cause the `LinkStorage` stops working. This changes will be pushed to this repository soon.

**Comment by aecio** [comment]
Already fixed in commit 0c066bc480c787cd1f2cb1d6d1eeb0f23db2a29a
